### ISSUE TITLE
Re-scan options

### ISSUE THREAD

**Issue by whyboris** [whyboris/Video-Hub-App#205]
### Catch-all of thoughts about re-scan options

⚠️ Check all re-scan code (since the extraction code was refactored recently).

⚠️ Make sure re-scan can happen quickly.

It's possible that users may not want to extract _clips_ into their _hub_, but later might change their mind. It would be nice to have a setting that they can edit _after_ the initial scan. This might reside in the _statistics_ at least for now. 

ps - consider adding into the _statistics_ the size of the hub on hard disk.

Perhaps it should be possible to change the number of screenshots to take per video after the initial scan. This too might reside in the _statistics_ tab in the settings menu.

_Maybe(?)_ include a toggle to re-scan the hub on-open (unsure whether this should be a global setting or a per-vha file (saved inside the `vha` file).

A user once requested a right-click 're-scan' though it's not clear whether that's needed once the general re-scan functionality is solid.

**Comment by cal2195** [comment]
> Perhaps it should be possible to change the number of screenshots to take per video after the initial scan.

If so, it would be nice to have the _option_ to regenerate all current clips to reflect the new setting! 👍

**Comment by whyboris** [comment]
#215 is a start of this saga 😅 

It first moves all the relevant methods inside `main.ts` to a single section and moves all related methods from `main-support.ts` into its own file `man-rescan.ts` 👍 

From there it will be easier to see what methods depend on what other methods, and whether there's any duplication of code (trying to keep things _DRY_).

After merging I'll take stock of what methods we already have and what functionality we are missing 🙌

**Comment by whyboris** [comment]
We currently have four re-scan buttons to click on:

https://github.com/whyboris/Video-Hub-App/blob/master/main.ts#L743

1. import-new-files
 - re-scan for all changes (addition, renaming, deleting)
2. rescan-current-directory
 - quick re-scan for only new videos
3. regenerate-library
 - regenerate all metadata except for thumbnails
4. verify-thumbnails
 - import any missing thumbnails

I'm a little tempted to comment out (3) for version 2 release -- it seems not needed unless the app starts tracking new metadata that was never extracted previously? @cal2195 -- no rush, but when you have a chance, could you comment on whether I understood (3) correctly, and perhaps share a use case other than the one I mentioned?

**Comment by cal2195** [comment]
(3) is also intended for the use case where (1) or (2) doesn't function correctly (as they are based on comparing current hdd contents with the in memory ones), so (3) does a clean scan of the hdd and ignores its in memory one for a more robust scan! :) Also very useful if some video contents change (eg. accidentally imported while downloading before), in case the other options don't pick up the new changes! 👍

**Comment by whyboris** [comment]
I'll think about this some more, but I think for version 2 release I'll keep (3) commented out for the release. Primarily I want to release version 2 ASAP so I'll spend time on testing everything and polishing things up (adding new icons, etc) -- so much to do 😱 

Once all that is done, before version 2 release I'll revisit whether to enable (3).

Additionally, at least for version 2.0.0 release, I don't think I'll let users change their preferences for the hub from the UI (that is, if they chose not to extract clips before, they will not be able to change that from within the app -- only by manually hacking the `.vha2` file). This is also done because I'd like to release sooner rather than later.

@cal2195 if you feel strongly that (3) should be in the app, just comment below and I'll coordinate with you before verison 2 release 👍 👌 🙇

**Comment by whyboris** [comment]
Related: #15

**Comment by cal2195** [comment]
I really do like (3), but I'm also open to a _yes I know what I'm doing give me all the options and enable everything plz and thx_ mode! 😅

**Comment by whyboris** [comment]
I guess it's easy enough to add the extra button and re-enable the functionality 👌 

I wish I had something in place that asked for a confirmation before the action occurred -- so if the user _accidentally_ clicks on one of the re-scan buttons, it doesn't mess up their hub 😅 

I'll see if it's easy enough -- I think I'll implement something 👍
